devices/src/tpm.rs

```diff
@@ -0,0 +1,439 @@
+use phf::{phf_map};
+use anyhow::anyhow;
+use thiserror::Error;
+use vtpm::tpm_emulator::{TPMBackendCmd, TPMEmulator};
+use vtpm::tpm_ioctl::TPM_CRB_BUFFER_MAX;
+use vm_device::BusDevice;
+use vtpm::tpm_ioctl::TPM_SUCCESS;
+use std::sync::{Arc, Barrier};
+use std::cmp;
+#[cfg(target_arch = "x86_64")]
+use arch::x86_64::layout::{VTPM_SIZE};
+#[cfg(target_arch = "aarch64")]
+use arch::aarch64::layout::{VTPM_START, VTPM_SIZE};
+
+
```

Comment:
I'm going to stop the review here since it's obvious you didn't do basic checks before submitting e.g. cargo fmt and clippy.